synapseclient/client.py

```diff
@@ -126,6 +126,12 @@
                          "max_wait"          : 30,
                          "back_off"          : 2}
 
+#Concrete types for UploadDestinations
+_SYNAPSE_S3_UPLOAD_DESTINATION_TYPE = 'org.sagebionetworks.repo.model.file.S3UploadDestination'
+_EXTERNAL_UPLOAD_DESTINATION_TYPE = 'org.sagebionetworks.repo.model.file.ExternalUploadDestination'
+_EXTERNAL_S3_UPLOAD_DESTINATION_TYPE = 'org.sagebionetworks.repo.model.file.ExternalS3UploadDestination'
+
```

Comment:
Why separate these as constants?  We have been using the concrete type all over the code before - is the plan that we define all of these as constants eventually?